tracing/middlewares.go

```diff
@@ -38,3 +38,34 @@ func InjectHTTPServerSpanWithTracer(name string, tracer *Tracer) endpoint.Middle
 		}
 	}
 }
+
+// InjectThriftServerSpan returns a go-kit endpoint.Middleware that injects a server
+// span into the `next` context.
+//
+// Starts the server span before calling the `next` endpoint and stops the span
+// after the endpoint finishes.
+// If the endpoint returns an error, that will be passed to span.Stop.
+//
+// Note, this depends on the edge context headers already being set on the
+// context object.  You can configure the thrift server to do this by passing
+// thriftbp.HeadersToForward to the server's SetForwardHeaders method.
+func InjectThriftServerSpan(name string) endpoint.Middleware {
+	return InjectThriftServerSpanWithTracer(name, nil)
+}
+
+// InjectThriftServerSpanWithTracer is the same as InjectThriftServerSpan except it
+// uses StartSpanFromThriftContextWithTracer to initialize the server span rather
+// than StartSpanFromThriftContext.
+func InjectThriftServerSpanWithTracer(name string, tracer *Tracer) endpoint.Middleware {
+	return func(next endpoint.Endpoint) endpoint.Endpoint {
+		return func(ctx context.Context, request interface{}) (response interface{}, err error) {
+			ctx, span := StartSpanFromThriftContextWithTracer(ctx, name, tracer)
+			defer func() {
+				span.Stop(ctx, err)
```

Comment:
Should we check the error here?

Reply:
good point. it might make sense to pass in a `log.Wrapper` and log the error here. but I won't worry about it too much (e.g. if it's too inconvenient to pass in `log.Wrapper` then I'm fine with ignoring the errors)

Reply:
`tracer` has a Logger :)